internal/http_server.go

```diff
@@ -650,9 +703,23 @@ func decisionForNginx2(
 	decisionListsMutex.Lock()
 	decision, ok = (*decisionLists).GlobalDecisionLists[clientIp]
 	decisionListsMutex.Unlock()
+	foundInIpFilter := false
 	if !ok {
-		// log.Println("no mention in global lists")
-	} else {
+		for _, iterateDecision := range []Decision{Allow, Challenge, NginxBlock, IptablesBlock} {
+			// check if Ipfilter ref associated to this iterateDecision exists
+			if _, globalOk := (*decisionLists).GlobalDecisionListsIPFilter[iterateDecision]; globalOk {
+				if (*decisionLists).GlobalDecisionListsIPFilter[iterateDecision].Allowed(clientIp) {
+					if config.Debug {
+						log.Printf("matched in ipfilter %v %s", iterateDecision, clientIp)
+					}
+					decision = iterateDecision
+					foundInIpFilter = true
+					break
```

Comment:
I haven't tested it but I think this `break` in line 717 could cause that specific rules are not applied in some corner cases, e.g. you define a generic rule to Allow 192.168.0.0/24 but another rule wants to block only 192.168.0.12

Reply:
OK great, so then I guess we can create a task for fixing it later, thank you Jeremy :)